### PR TITLE
Allow default construction of tasks with FINISHED goal state

### PR DESCRIPTION
**Pull request by gabrielhartmann**

### CODE REVIEW

**Comment by gabrielhartmann** [comment]
@tobilg:  This should resolve the issue you were having with MongoDB authoring.

**Comment by tobilg** [comment]
I implemented it slightly different, but I can confirm that my changes have the desired effect:

* https://github.com/tobilg/dcos-commons/blob/mongodb-replicaset/sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultPlanGenerator.java#L63
* https://github.com/tobilg/dcos-commons/blob/mongodb-replicaset/sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultPhaseFactory.java#L55

**Comment by gabrielhartmann** [comment]
@mesosphere-ci: retest this please

File: frameworks/hdfs/integration/tests/test_upgrade.py [hunk 1]
```diff
@@ -24,10 +24,6 @@ def setup_module(module):
     uninstall()
 
 
-def teardown_module(module):
```
Comment (nickbp) [inline frameworks/hdfs/integration/tests/test_upgrade.py]: any reason to remove these? doesn't hurt to clean up after tests are done?

**Comment by nickbp** [comment]
Question answered offline, LGTM (answer was: ability to diagnose still-running service after failed test)

File: frameworks/hdfs/integration/tests/test_upgrade.py (hunk 1 above)
Comment (nickbp) [inline frameworks/hdfs/integration/tests/test_upgrade.py]: Question answered offline, LGTM (answer was: ability to diagnose still-running service after failed test)

Comment (gabrielhartmann) [inline frameworks/hdfs/integration/tests/test_upgrade.py]: I want to go look at the HDFS cluster as it's running in a failing test scenario.

**Pull request merged**